From f76ad8ec4c4fdf8f8f9ba20d969306bba8fe26cd Mon Sep 17 00:00:00 2001
From: William Lai <b04597@freescale.com>
Date: Mon, 25 Jan 2010 15:32:26 +0800
Subject: [PATCH] ENGR00120372 CAN: Clock source for FlexCAN module can not be changed

Change the function strcasecmp(buf, "osc") to strncasecmp(buf, "osc",
3), which only compare the first 3bytes.

Signed-off-by: William Lai <b04597@freescale.com>
---
 drivers/net/can/flexcan/dev.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/can/flexcan/dev.c b/drivers/net/can/flexcan/dev.c
index f8386e5..cc117f0 100644
--- a/drivers/net/can/flexcan/dev.c
+++ b/drivers/net/can/flexcan/dev.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2008-2009 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2008-2010 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -338,9 +338,9 @@ static ssize_t flexcan_set_attr(struct device *dev,
 		goto set_finish;
 
 	if (attr_id == FLEXCAN_ATTR_BR_CLKSRC) {
-		if (!strcasecmp(buf, "bus"))
+		if (!strncasecmp(buf, "bus", 3))
 			flexcan->br_clksrc = 1;
-		else if (!strcasecmp(buf, "osc"))
+		else if (!strncasecmp(buf, "osc", 3))
 			flexcan->br_clksrc = 0;
 		goto set_finish;
 	}
-- 
1.5.4.4

